nx-X11/programs/Xserver/hw/nxagent/Keyboard.c

```diff
@@ -923,6 +923,14 @@ N/A
         fprintf(stderr, "nxagentKeyboardProc: nxagentKeyboard is [%s].\n", nxagentKeyboard ? nxagentKeyboard : "NULL");
         #endif
 
+        if (nxagentKeyboard && (strcmp(nxagentKeyboard, "null/null") == 0))
+        {
+          #ifdef TEST
+          fprintf(stderr, "nxagentKeyboardProc: changing nxagentKeyboard from [null/null] to [clone].\n");
+          #endif
+          free(nxagentKeyboard);
+          nxagentKeyboard = strdup("clone");
```

Comment:
no, I don't want to use update_string() outside Keyboard.c. Reconnect.c has very similar code. Soi I prefer to let it this way. Besides, nxagentKeyboard is a global variable and update_string ist only using variables local to Keyboard.c